### PR TITLE
Energy field fix

### PR DESCRIPTION
**Pull request by NotStanley4330**
Fix for [#268](https://github.com/exult/exult/issues/268). This extends the functionality of ```sfxinc.h``` and the ```shape_info.txt``` file by adding a halt_frame to the possible fields in ```%%section shape_sfx``` . In order for this frame to be used and halt playing the sound effect I created a simple inline function to compare the shapes current frame against the listed halt frame. If the current frame is the halt frame, then the sfx should halt.

### CODE REVIEW

File: objs/animate.cc [hunk 1]
```diff
@@ -59,6 +59,22 @@ static inline bool Get_sfx_out_of_range(
 		   > (MAX_SOUND_FALLOFF * MAX_SOUND_FALLOFF);
 }
 
+static inline bool Is_halt_frame(
```
Comment (marzojr) [inline objs/animate.cc:62]: Please run clang-format on this file.

**Review by marzojr — CHANGES_REQUESTED** [review]

**Comment by marzojr** [comment]
Hm, see also my comment on the issue thread.

File: objs/animate.cc (hunk 1 above)
Comment (NotStanley4330) [reply objs/animate.cc]: I think I ran clang-format correctly. Please let me know if it didn't reformat correctly.

**Comment by NotStanley4330** [comment]
I am going to close this as I have replaced it with an entirely different solution

**Pull request closed without merging**